### PR TITLE
RevEng: Allow null as default value

### PR DESCRIPTION
**Pull request by lajones**
Fix for #3248 and #3249.

### CODE REVIEW

**Comment by ajcvickers** [comment]
I tried to review this but I'm struggling to understand the scope. Do we only do the DEFAULT(NULL) if the application explicitly sets "NULL" as the default value SQL? Or do we also do it if CLR default value is set to null? And what f the CLR default is not set, but the type is nullable? And for that matter, can we distinguish between setting null and it just being null because it wasn't set?

It seems to me that the only time we should create DEFAULT(NULL) is when the application has explicitly set the default value SQL to "NULL", and even that case is a little dubious. In all other cases we should just not set the default for nullable columns because it will be null anyway.

**Comment by lajones** [comment]
@ajcvickers I think you have this the wrong way round? This is not a case of the application affecting the metadata of the database but the other way around. Previously, if a user created a column in their database and put `DEFAULT (NULL)` in their column definition, when Reverse Engineering tried to understand that default value in order to generate the appropriate `HasDefaultValue()` calls it would issue a warning that we didn't understand the value `(NULL)`. This fix updates the code so we no longer output that warning.

Following our last conversation my understanding was that there was never a need to output a `HasDefaultValue(null)` in the generated code because in order for the database to allow that situation the column must be nullable and the default value for any nullable type that we generate is null anyway. So in the expected results for the added test columns note that they do _not_ have that API call. But please let me know if I misunderstood.

**Comment by ajcvickers** [comment]
@lajones Okay, that makes sense. Let me look again.

File: src/EntityFramework.SqlServer.Design/Utilities/SqlServerLiteralUtilities.cs [hunk 1]
```diff
@@ -121,6 +121,15 @@ public virtual string ConvertSqlServerStringLiteral([NotNull] string sqlServerSt
                 };
             }
 
+            if (propertyType.IsNullableType()
+                && "NULL" == sqlServerDefaultValue)
```
Comment (ajcvickers) [inline src/EntityFramework.SqlServer.Design/Utilities/SqlServerLiteralUtilities.cs]: Is it always uppercase NULL? Also, the style we use everywhere else is to put the string literal on the right of the operator.

**Comment by ajcvickers** [comment]
:shipit:

File: src/EntityFramework.SqlServer.Design/Utilities/SqlServerLiteralUtilities.cs (hunk 1 above)
Comment (lajones) [inline src/EntityFramework.SqlServer.Design/Utilities/SqlServerLiteralUtilities.cs]: Thanks. Yes, it is capitalized even if you put some different capitalization in the column definition. I'll change the order of the comparison.

**Comment by lajones** [comment]
Checked in with commit 8600030.

**Pull request closed without merging**